New warning in gameworld.cpp

```
[2/4] Building CXX object CMakeFiles/DDNet.dir/src/game/client/prediction/gameworld.cpp.o
src/game/client/prediction/gameworld.cpp: In member function ‘void CGameWorld::CreateExplosion(vec2, int, int, bool, int, int64_t)’:
src/game/client/prediction/gameworld.cpp:305:23: warning: ‘#‘target_mem_ref’ not supported by dump_expr#<expression error>’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   vec2 Diff = apEnts[i]->m_Pos - Pos;
```
@trml

--- a/src/game/client/prediction/entities/character.cpp
+++ b/src/game/client/prediction/entities/character.cpp
@@ -1071,15 +1071,12 @@ void CCharacter::Read(CNetObj_Character *pChar, CNetObj_DDNetCharacter *pExtende
 		SetActiveWeapon(pChar->m_Weapon);
 	}
 
-	if(!pExtended)
+	if(pChar->m_Emote != EMOTE_PAIN && pChar->m_Emote != EMOTE_NORMAL)
+		m_DeepFreeze = false;
+	if(pChar->m_Weapon != WEAPON_NINJA || pChar->m_AttackTick > m_FreezeTick || absolute(pChar->m_VelX) == 256*10)
 	{
-		if(pChar->m_Emote != EMOTE_PAIN && pChar->m_Emote != EMOTE_NORMAL)
-			m_DeepFreeze = false;
-		if(pChar->m_Weapon != WEAPON_NINJA || pChar->m_AttackTick > m_FreezeTick || absolute(pChar->m_VelX) == 256*10)
-		{
-			m_DeepFreeze = false;
-			UnFreeze();
-		}
+		m_DeepFreeze = false;
+		UnFreeze();
 	}
 	if(!GameWorld()->m_WorldConfig.m_PredictFreeze)
 	{
@@ -1152,13 +1149,6 @@ void CCharacter::Read(CNetObj_Character *pChar, CNetObj_DDNetCharacter *pExtende
 		m_Core.m_Hook = !(pExtended->m_Flags & CHARACTERFLAG_NO_HOOK);
 		m_SuperJump = pExtended->m_Flags & CHARACTERFLAG_ENDLESS_JUMP;
 		m_Jetpack = pExtended->m_Flags & CHARACTERFLAG_JETPACK;
-
-		if(GameWorld()->m_WorldConfig.m_PredictFreeze)
-			m_DeepFreeze = pExtended->m_Flags & CHARACTERFLAG_DEEP_FROZEN;
-
-		bool Frozen = pExtended->m_Flags & CHARACTERFLAG_FROZEN;
-		if(m_FreezeTime > 0 && !Frozen)
-			UnFreeze();
 
 		m_Hit = HIT_ALL;
 		if(pExtended->m_Flags & CHARACTERFLAG_NO_GRENADE_HIT)

--- a/src/game/client/prediction/gameworld.cpp
+++ b/src/game/client/prediction/gameworld.cpp
@@ -58,6 +58,8 @@ CEntity *CGameWorld::FindLast(int Type)
 
 int CGameWorld::FindEntities(vec2 Pos, float Radius, CEntity **ppEnts, int Max, int Type)
 {
+	if(!ppEnts)
+		return 0;
 	if(Type < 0 || Type >= NUM_ENTTYPES)
 		return 0;
 
@@ -66,8 +68,7 @@ int CGameWorld::FindEntities(vec2 Pos, float Radius, CEntity **ppEnts, int Max, 
 	{
 		if(distance(pEnt->m_Pos, Pos) < Radius+pEnt->m_ProximityRadius)
 		{
-			if(ppEnts)
-				ppEnts[Num] = pEnt;
+			ppEnts[Num] = pEnt;
 			Num++;
 			if(Num == Max)
 				break;
@@ -296,12 +297,14 @@ CEntity *CGameWorld::GetEntity(int ID, int EntType)
 void CGameWorld::CreateExplosion(vec2 Pos, int Owner, int Weapon, bool NoDamage, int ActivatedTeam, int64_t Mask)
 {
 	// deal damage
-	CCharacter *apEnts[MAX_CLIENTS];
+	CCharacter *apEnts[MAX_CLIENTS] = {0};
 	float Radius = 135.0f;
 	float InnerRadius = 48.0f;
 	int Num = FindEntities(Pos, Radius, (CEntity**)apEnts, MAX_CLIENTS, CGameWorld::ENTTYPE_CHARACTER);
 	for(int i = 0; i < Num; i++)
 	{
+		if(!apEnts[i])
+			continue;
 		vec2 Diff = apEnts[i]->m_Pos - Pos;
 		vec2 ForceDir(0,1);
 		float l = length(Diff);

--- a/src/game/client/prediction/gameworld.cpp
+++ b/src/game/client/prediction/gameworld.cpp
@@ -303,8 +303,6 @@ void CGameWorld::CreateExplosion(vec2 Pos, int Owner, int Weapon, bool NoDamage,
 	int Num = FindEntities(Pos, Radius, (CEntity**)apEnts, MAX_CLIENTS, CGameWorld::ENTTYPE_CHARACTER);
 	for(int i = 0; i < Num; i++)
 	{
-		if(!apEnts[i])
-			continue;
 		vec2 Diff = apEnts[i]->m_Pos - Pos;
 		vec2 ForceDir(0,1);
 		float l = length(Diff);

